test/ParseableInterface/ModuleCache/RebuildRemarks/missing-dependency.swift

```diff
@@ -0,0 +1,26 @@
+// RUN: %empty-directory(%t/ModuleCache)
+// RUN: %empty-directory(%t/Build)
+
+// 1. Create a dummy module
+// RUN: echo 'public func publicFunction() {}' > %t/TestModule.swift
+
+// 2. Create both an interface and a compiled module for it
+// RUN: %target-swift-frontend -emit-module -o %t/Build/TestModule.swiftmodule %t/TestModule.swift -emit-module-interface-path %t/Build/TestModule.swiftinterface -swift-version 5 -module-name TestModule
+
+// 3. Make the compiled module unreadable so it gets added as a dependency but is not used
+// RUN: echo ' ' | cat - %t/Build/TestModule.swiftmodule > %t/Build/TestModule.swiftmodule.edited
+// RUN: mv %t/Build/TestModule.swiftmodule.edited %t/Build/TestModule.swiftmodule
+
+// 4. Try to import the module, which will create a cached module that depends on both the .swiftmodule and .swiftinterface
+// RUN: %target-swift-frontend -typecheck %s -I %t/Build -module-cache-path %t/ModuleCache -Xllvm -debug-only=textual-module-interface
+
+// 5. Remove the .swiftmodule, which will result in a missing dependency note
```

Comment:
I think the test should test the expected use case, which is that when you put the module back the way it was, the interface isn't used despite there being an entry in the cache.